feat: Add migration for food table

emmanuel-carreira opened the pull request:
1. Add food table creation and deletion.

2. Add meal_food table to relate meals and foods.

emmanuel-carreira committed:
feat: Add migration for food table

1. Add food table creation and deletion

2. Add meal_food table to relate meals and foods

emmanuel-carreira commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Create food table migration file](https://trello.com/c/r75l5ctz/7-create-food-table-migration-file)

viiniciuspaes reviewed (changes requested):
I think it`s better to maintainability if we don't use the same migration for more than one table. This is crating a chain dependence that can lead to problems, if food table for some reason needs to be rewrite we will have to migrate meal table as well and vice versa.

emmanuel-carreira commented:
If i understand correctly, you're talking about the meal table and not the meal_food table that is used to relate the meal table and the food table, right? If so, the meal table is in another migration file. The migration file in this PR is the one that creates/drops the food and the meal_food tables.

I talked with @FernandoCCNeto about it and we concluded it was better to use only one file to create/drop both of these tables in case we need to rollback some migration. This way we avoid a corrupted database that may have the meal_food table and not the food table (so the foreign key in the meal_food table points to an database entry that does not exists). 

But thinking about your point, i guess it's better to use two migration files, each one with it's own table. Using the food table migration prior to the meal_food migration guarantees that no such corrupted database will exist.

emmanuel-carreira committed:
refactor: Move the meal_food table to it's own migration file

1. Delete the meal_food table creation/deletion from the food migration file.

2. Add a new migration file containing the meal_food table migration.
